ui-v2/src/api/flow-runs/index.ts

```diff
@@ -217,3 +217,93 @@ export const useDeploymentCreateFlowRun = () => {
 		...rest,
 	};
 };
+
+/**
+ * Request body for setting a flow run state
+ */
+type SetFlowRunStateParams = {
+	/**
+	 * The ID of the flow run to update
+	 */
+	id: string;
+	/**
+	 * The new state type to set
+	 */
+	state: components["schemas"]["StateType"];
+	/**
+	 * Optional name for the state
+	 */
+	name?: string | null;
+	/**
+	 * Optional message to associate with the state change
+	 */
+	message?: string | null;
+	/**
+	 * Whether to force the state change, bypassing orchestration rules
+	 */
+	force?: boolean;
+};
+
+/**
+ * Hook for changing a flow run's state
+ *
+ * @returns Mutation object for setting a flow run state with loading/error states and trigger function
+ *
+ * @example
+ * ```ts
+ * const { setFlowRunState, isLoading } = useSetFlowRunState();
+ *
+ * setFlowRunState({
+ *   id: "flow-run-id",
+ *   state: "COMPLETED",
+ *   name: "Custom state name",
+ *   message: "State changed by user"
+ * }, {
+ *   onSuccess: () => {
+ *     console.log('Flow run state changed successfully');
+ *   },
+ *   onError: (error) => {
+ *     console.error('Failed to change flow run state:', error);
+ *   }
+ * });
+ * ```
+ */
+export const useSetFlowRunState = () => {
+	const queryClient = useQueryClient();
+	const { mutate: setFlowRunState, ...rest } = useMutation({
+		mutationFn: async ({
+			id,
+			state,
+			name,
+			message,
+			force = false,
+		}: SetFlowRunStateParams) => {
+			const res = await getQueryService().POST("/flow_runs/{id}/set_state", {
+				params: { path: { id } },
+				body: {
+					state: {
+						type: state,
+						name: name,
+						message: message,
+					},
+					force,
+				},
+			});
+
+			if (!res.data) {
+				throw new Error("'data' expected");
+			}
+			return res.data;
+		},
+		onSuccess: () => {
+			// After a successful state change, invalidate all flow run queries to refetch
+			return queryClient.invalidateQueries({
+				queryKey: queryKeyFactory.all(),
+			});
```

Comment:
Chances are we'll need to invalidate a detail query for the flow run too, but it looks like that hasn't been created yet. This seems like a good candidate for an [optimistic update](https://tanstack.com/query/v4/docs/framework/react/guides/optimistic-updates).